.yarn/patches/@docusaurus-mdx-loader-npm-2.4.3-0d95b4751a.patch

```diff
@@ -0,0 +1,64 @@
+\ No newline at end of file
++{"version":3,"file":"loader.d.ts","sourceRoot":"","sources":["../src/loader.ts"],"names":[],"mappings":"AAAA;;;;;GAKG;AAuBH,OAAO,KAAK,EAAC,cAAc,EAAC,MAAM,mBAAmB,CAAC;AACtD,OAAO,KAAK,EAAC,aAAa,EAAC,MAAM,SAAS,CAAC;AAC3C,OAAO,KAAK,EAAY,MAAM,EAAC,MAAM,SAAS,CAAC;AAC/C,OAAO,KAAK,EAAC,iBAAiB,EAAC,MAAM,sBAAsB,CAAC;AAsB5D,oBAAY,SAAS,GAEnB;IAAC,MAAM,CAAC,GAAG,EAAE,CAAC;IAAE,GAAG;CAAC,GAAG,MAAM,CAAC,GAAG,EAAE,CAAC,CAAC;AAEvC,oBAAY,UAAU,GAAG;IACvB,WAAW,EAAE,OAAO,GAAG,iBAAiB,CAAC;IACzC,aAAa,EAAE,SAAS,EAAE,CAAC;IAC3B,aAAa,EAAE,SAAS,EAAE,CAAC;IAC3B,0BAA0B,EAAE,SAAS,EAAE,CAAC;IACxC,0BAA0B,EAAE,SAAS,EAAE,CAAC;CACzC,CAAC;AAEF,oBAAY,OAAO,GAAG,OAAO,CAAC,UAAU,CAAC,GAAG;IAC1C,cAAc,EAAE,cAAc,CAAC;IAC/B,UAAU,EAAE,MAAM,EAAE,CAAC;IACrB,OAAO,EAAE,MAAM,CAAC;IAChB,YAAY,CAAC,EAAE,CAAC,QAAQ,EAAE,MAAM,KAAK,OAAO,CAAC;IAC7C,sCAAsC,CAAC,EAAE,OAAO,CAAC;IACjD,kBAAkB,CAAC,EAAE,OAAO,CAAC;IAC7B,YAAY,CAAC,EAAE,MAAM,GAAG,CAAC,CAAC,QAAQ,EAAE,MAAM,KAAK,MAAM,CAAC,CAAC;IACvD,YAAY,CAAC,EAAE,CAAC,QAAQ,EAAE;QACxB,WAAW,EAAE;YAAC,CAAC,GAAG,EAAE,MAAM,GAAG,OAAO,CAAA;SAAC,CAAC;QACtC,QAAQ,EAAE;YAAC,CAAC,GAAG,EAAE,MAAM,GAAG,OAAO,CAAA;SAAC,CAAC;KACpC,KAAK;QAAC,CAAC,GAAG,EAAE,MAAM,GAAG,OAAO,CAAA;KAAC,CAAC;CAChC,CAAC;AA8EF,wBAAsB,SAAS,CAC7B,IAAI,EAAE,aAAa,CAAC,OAAO,CAAC,EAC5B,UAAU,EAAE,MAAM,GACjB,OAAO,CAAC,IAAI,CAAC,CA+Hf"}
+\ No newline at end of file
+diff --git a/lib/loader.js b/lib/loader.js
+index 5f9dce1f556491abab5b8307658b517f314ed983..07f9b13dc27730cb3093c69a43a48bb68eb66089 100644
+--- a/lib/loader.js
++++ b/lib/loader.js
+@@ -8,6 +8,7 @@
+ Object.defineProperty(exports, "__esModule", { value: true });
+ exports.mdxLoader = void 0;
+ const tslib_1 = require("tslib");
++const url_1 = require("url");
+ const fs_extra_1 = tslib_1.__importDefault(require("fs-extra"));
+ const logger_1 = tslib_1.__importDefault(require("@docusaurus/logger"));
+ const utils_1 = require("@docusaurus/utils");
+@@ -200,8 +201,8 @@ ${assets ? `export const assets = ${createAssetsExportCode(assets)};` : ''}
+ `;
+     const code = `
+ ${pragma}
+-import React from 'react';
+-import { mdx } from '@mdx-js/react';
++import React from "${(0, url_1.pathToFileURL)(require.resolve('react'))}";
++import { mdx } from "${(0, url_1.pathToFileURL)(require.resolve('@mdx-js/react'))}";
```

Comment:
Would this allow removing the dependencies added in https://github.com/yarnpkg/berry/pull/5625/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519?

If you could send a PR upstream to fix this as well that would be great.

Reply:
> Would this allow removing the dependencies added in https://github.com/yarnpkg/berry/pull/5625/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519?

Hmm.. Probably. I will test later

> If you could send a PR upstream to fix this as well that would be great.

I will submit one. Though I do think this code pattern is kinda obscure and won't be surprised if it is rejected.

Reply:
@josh-cena Indicated that they would accept such a change in https://discord.com/channels/226791405589233664/654372321225605128/949824347529445487

Reply:
By the way, Docusaurus now uses MDX v3 and React automatic runtime, so things may be different ;)

Reply:
Yeah. Last night I started looking into making a PR and found that part of the MDX loader in Docusaurus v3 has changed significantly.

@merceyz Let's wait until after we migrate to Docusaurus v3 to see whether we still need the patch and/or to make an upstream PR